frontend/catalyst/pennylane_extensions.py

```diff
@@ -242,22 +245,44 @@ def default_expand_fn(self, circuit, max_expansion=10):
         if any(isinstance(op, MidMeasureMP) for op in circuit.operations):
             raise CompileError("Must use 'measure' from Catalyst instead of PennyLane.")
 
+        # TODO: At the moment, lightning.qubit and other lightning devices are special because
+        # backend_name is indeed a name. However, for non-lightning devices, backend_name is
+        # actually a path to a shared library.
+        # So, let's try to find the path to this configuration file based on the name.
+        # The assumption is that it has to live in the same directory as the C library.
+        def get_spec_path(name, name_or_path_str):
+            name_or_path = pathlib.Path(name_or_path_str)
+            path = name_or_path.parent
+            path = path / "backend" / (name + ".toml")
+            return path
+
+        spec_path = get_spec_path(self.backend_name_canonical, self.backend_lib)
+        with open(spec_path, "rb") as f:
+            spec = toml_load(f)
+
+        # At the moment assume QJIT Device's decomposition logic is guided by the specification of
+        # lightning.qubit. All "full" gates are allowed. All "matrix" gates are decomposed to
+        # to qml.QubitUnitary
+        decompose_to_qubit_unitary = spec["operations"]["gates"][0]["matrix"]
```

Comment:
I am not sure how I feel about this. Again, this relates a bit to decomposition, but aren't there an infinite number of gates? Wouldn't it be best to define these matrix gates as "all gates which are not full gates"? But then going back to decomposition, how do we know whether we apply some other decomposition rule or just decompose it into `qml.QubitUnitary`?

Reply:
:+1:

Reply:
Another addendum, because Erick discovered that replacing a gate with QubitUnitary will make the circuit non-differentiable, is that we could detect whether `grad` is used on a QNode. In that case, we could overwrite the non-native strategy expressed by the device to be decomp-only, and in all other cases adhere to the device preference. Maybe this could be a followup PR though.

Reply:
I added the issue here: https://github.com/PennyLaneAI/catalyst/issues/398